src/components/message/view.js

```diff
@@ -89,7 +89,7 @@ export class QuotedMessage extends React.Component<
       if (props.message.messageType === 'media') return false;
       const jsonBody = JSON.parse(props.message.content.body);
       return (
-        !jsonBody.blocks.length > 1 ||
+        jsonBody.blocks.length < 1 ||
```

Comment:
Shouldn't this be `jsonBody.blocks.length <= 1`?

Reply:
@mxstbr i think it's not correct, here code section slightly complicated.
i'm afraid after too long explanation...

## Explain about `props.message.content.body`

original passed prop value `JSON.parse(props.message.content.body)`(line 90) is `string` type, but actually serialized `Object`.

there is actual value of passed to `JSON.parse()`(sorry about JP char😇)
　  
![screen shot 2018-05-14 at 20 01 18](https://user-images.githubusercontent.com/5501268/39994077-1d93195a-57b3-11e8-9f2a-2f79ac015c42.png)

and finally UI view on browser.
　  
![screen shot 2018-05-14 at 20 00 53](https://user-images.githubusercontent.com/5501268/39994177-724865ae-57b3-11e8-9816-87ee890d9b65.png)

- type definition in `shared/graphql/fragments/message/messageInfo.js`
```js
export type MessageInfoType = {
  id: string,
  timestamp: Date,
  messageType: string,
  parent: {
    id: string,
    timestamp: Date,
    messageType: string,
    author: {
      ...$Exact<ThreadParticipantType>,
    },
    content: {
      body: string,
    },
  },
  author: {
    ...$Exact<ThreadParticipantType>,
  },
  reactions: {
    count: number,
    hasReacted: boolean,
  },
  content: {
    body: string, /** @see ⬅︎ string, but serialize Object */
  },
};
```
- original `props.message.content.body` value

```
"{\"entityMap\":{},\"blocks\":[{\"depth\":0,\"type\":\"unstyled\",\"text\":\"あゆあや\",\"entityRanges\":[],\"inlineStyleRanges\":[]}]}"
```

- `JSON.parse()` ed `props.message.content.body` value

```
{
  "entityMap": {},
  "blocks": [
    {
      "depth": 0,
      "type": "unstyled",
      "text": "あゆあや",
      "entityRanges": [],
      "inlineStyleRanges": []
    }
  ]
}
```

## Conclusion
`jsonBody.blocks` is `Array`, and contain following shape.

```js
  "blocks": [
    {
      "depth": 0,
      "type": "unstyled",
      "text": "あゆあや",
      "entityRanges": [],
      "inlineStyleRanges": []
    }
  ]
```

therefore `jsonBody.blocks.length` is always 1(if it is persisted as a correct DraftJS json), doesn't matter **is the message text short or not?**, i think😀

but i just also find out very closely code blocks at https://github.com/withspectrum/spectrum/blob/alpha/api/mutations/message/addMessage.js#L71-L74

should we write similar validate code even here?🤔

message text is spectrum's core element as a product, so i think might be better straightforward way for handle `message.content.body`.(I'd like to think refactor proposal when I have time!)

Reply:
> therefore jsonBody.blocks.length is always 1(if it is persisted as a correct DraftJS json), doesn't matter is the message text short or not?, i think

This is incorrect :blush: Try writing a message with a code block! `blocks.length` will be 2. What I'm trying to check here is whether the message has a code block (or two text blocks), at which point it will be longer than the limit already even if the text is short, so this should be `blocks.length <= 1`.

We don't need any validation code on the client since the server only allows DraftJS input anyway. I've also got some further enforcement improvements upon that in #2792.

Reply:
@mxstbr i fixed it😀

totally it's my fault and sorry about my lack of confirm🙇‍♂️
i try writing a message with a code block & JSON.parse()ed, i understand that fact.

when sending such a pullrequest from now on, keep track of the context of code and add unit testing as much as possible to try to reduce the burden on the reviewer🙏

> We don't need any validation code on the client since the server only allows DraftJS input anyway. I've also got some further enforcement improvements upon that in #2792.

it definitely valuable! it'll be nice work😀

Reply:
Hey, definitely no worries! Thanks for digging into this!